### PR TITLE
Upgrade JavaScript dependencies

### PR DESCRIPTION
**Pull request by nwalters512**
I skipped a) client-side packages and b) the majority of the major-version bumps, many of which are ESM-only.

### CODE REVIEW

File: apps/prairielearn/package.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
     "crypto-js": "^4.1.1",
     "csv-parse": "^5.4.0",
     "csvtojson": "^2.0.10",
-    "d3": "^7.8.4",
+    "d3": "^7.8.5",
```
Comment (jonatanschroeder) [inline apps/prairielearn/package.json:73]: I believe this is client-side (though I don't see an issue with updating it).

File: packages/prettier-plugin-sql/package.json [hunk 2]
```diff
@@ -13,11 +13,11 @@
...
-    "@types/node": "^18.16.16",
-    "typescript": "^5.1.3"
+    "@types/node": "^18.16.19",
+    "typescript": "^5.1.6"
   },
   "dependencies": {
     "@types/prettier": "^2.7.3",
-    "sql-formatter": "^12.2.1"
+    "sql-formatter": "^12.2.3"
   }
```
Comment (jonatanschroeder) [inline packages/prettier-plugin-sql/package.json:22]: Pending merge with Prettier 3...

**Review by jonatanschroeder — APPROVED** [review]
No issues that I can see.

File: apps/prairielearn/package.json (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/package.json:73]: True, it is client-side, though this is an incredibly stable package - like you said, it should be safe to do this.

File: packages/prettier-plugin-sql/package.json (hunk 2 above)
Comment (nwalters512) [reply packages/prettier-plugin-sql/package.json]: ✅

**Pull request merged**